tests/testthat/test-class.R

```diff
@@ -115,6 +115,18 @@ test_that("characters are handled correctly", {
 
 # Infrastructure ---------------------------------------------------------------
 
+test_that("bake method errors when needed non-standard role columns are missing", {
+  rec <- recipe(x) %>%
+    check_class(x1, x2) %>%
+    update_role(x1, new_role = "potato") %>%
+    update_role_requirements(role = "potato", bake = FALSE)
```

Comment:
It is a little confusing, what happens here is that `update_role_requirements(bake = FALSE)` stops `bake()` from erroring, so that we can check that `bake.check_class()` errors correctly.